### PR TITLE
fix(uploads): allow images/video/audio in mothership presigned route

### PR DESCRIPTION
**Pull request by waleedlatif1**
## Summary
- The mothership branch of `/api/files/presigned` calls `validateFileType`, which only permits `SUPPORTED_DOCUMENT_EXTENSIONS`. That rejects PNG screenshots (and any image/video/audio) at the new presigned upload path introduced in #4509, even though Mothership has always supported media attachments via the legacy `/api/files/upload` route.
- Introduce `validateAttachmentFileType` in `lib/uploads/utils/validation.ts`, backed by the union of document, code, image, audio, and video extension sets, and wire it into the mothership branch. Behavior now matches what `/api/files/upload` already enforced.
- Executables and unknown extensions are still rejected.

## Test plan
- [x] Drag a `.png` screenshot into Mothership chat input on staging — uploads successfully
- [x] Drag a `.mp4` / `.mp3` — uploads successfully
- [x] Drag a `.exe` — rejected with a clear error
- [x] `bun run check:api-validation` passes

### CODE REVIEW

File: apps/sim/lib/uploads/utils/validation.ts [hunk 1]
```diff
@@ -219,6 +219,37 @@ export interface FileValidationError {
   supportedTypes: string[]
 }
 
+export const SUPPORTED_ATTACHMENT_EXTENSIONS = [
+  ...SUPPORTED_DOCUMENT_EXTENSIONS,
+  ...SUPPORTED_CODE_EXTENSIONS,
+  ...SUPPORTED_IMAGE_EXTENSIONS,
+  ...SUPPORTED_AUDIO_EXTENSIONS,
+  ...SUPPORTED_VIDEO_EXTENSIONS,
+] as const
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/lib/uploads/utils/validation.ts]: Fixed in 93b15656 — `SUPPORTED_ATTACHMENT_EXTENSIONS` is now built via `Array.from(new Set([...]))` so `webm` appears once. Also added a unit test guarding against future duplicates.

File: apps/sim/app/api/files/presigned/route.ts [hunk 2]
```diff
@@ -141,7 +141,7 @@ export const POST = withRouteHandler(async (request: NextRequest) => {
         )
       }
 
-      const fileValidationError = validateFileType(fileName, contentType)
+      const fileValidationError = validateAttachmentFileType(fileName)
       if (fileValidationError) {
         throw new ValidationError(fileValidationError.message)
       }
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/app/api/files/presigned/route.ts:147]: Acknowledged — this is intentional. The legacy `/api/files/upload` route also validates by extension only (see `validateFileExtension` in `apps/sim/app/api/files/upload/route.ts`), so dropping MIME cross-validation here restores parity with the pre-#4509 behavior. The presigned URL is opaque to MIME mismatches anyway; serving routes derive content-type from the stored object metadata.

File: apps/sim/lib/uploads/utils/validation.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/sim/lib/uploads/utils/validation.ts]: Fixed in 93b15656 — `apps/sim/app/api/files/upload/route.ts` now imports `SUPPORTED_ATTACHMENT_EXTENSIONS` from `@/lib/uploads/utils/validation` instead of redefining the same union locally. Single source of truth for both upload paths.

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review
